quote/gap.go

```diff
@@ -0,0 +1,147 @@
+package quote
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/go-gota/gota/dataframe"
+	lib "github.com/wopta/goworkspace/lib"
+	"github.com/wopta/goworkspace/models"
+)
+
+func GapFx(w http.ResponseWriter, r *http.Request) (string, interface{}, error) {
+	req := lib.ErrorByte(io.ReadAll(r.Body))
+	var data models.Policy
+	defer r.Body.Close()
+	e := json.Unmarshal(req, &data)
+
+	authToken, err := models.GetAuthTokenFromIdToken(r.Header.Get("Authorization"))
+	lib.CheckError(err)
+	res, e := Gap(authToken.Role, data)
+	s, e := json.Marshal(res)
+	return string(s), nil, e
+
+}
+
+func Gap(role string, data models.Policy) (models.Policy, error) {
```

Comment:
We could pass the policy (data) as a pointer as to modify incrementally the same object (this is also valid for the function setOfferPrices)